wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/PrunerTest.kt

```diff
@@ -1462,6 +1462,54 @@ class PrunerTest {
     assertThat(message.imports).containsExactly("title.proto")
   }
 
+  /**
+   * We had a bug in import pruning where we retained imports if the files were non-empty,
+   * even if those imports were unnecessary.
+   */
+
+  @Test
+  fun importPruningIsPrecise() {
+    val schema = buildSchema {
+      add(
+        "message.proto".toPath(),
+        """
+             |import 'footer.proto';
```

Comment:
Previously this import was being retained event the `AnotherMessage` was being pruned, along with the need to keep importing it.